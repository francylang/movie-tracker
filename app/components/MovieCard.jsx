import React from 'react';
import PropTypes from 'prop-types';

const MovieCard = ({ title, releaseDate, summary, score, img }) => {

  let poster = `https://image.tmdb.org/t/p/w500/${img}`;

  return (
    <div className='movie-card'>
      <h2>{title}</h2>
      <div className="fav-btn"></div>
      <div className="movie-container">
        <img className="movie-image" src={poster} alt="poster" />
        <div className="score-release-container">
          <h4>Release Date</h4>
          <h3 className="release">{releaseDate}</h3>
          <h4 className="rating">User Rating</h4>
          <h3>{score}</h3>
        </div>
      </div>
      <p>{summary}</p>
    </div>
  );
};

MovieCard.propTypes = {
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  summary: PropTypes.string,
  score: PropTypes.number,
  img: PropTypes.string
};

export default MovieCard;


// let backDrop = `https://image.tmdb.org/t/p/w500${poster_path}`;
// <img className="movie-image" src={backDrop} alt="backdrop image" />
