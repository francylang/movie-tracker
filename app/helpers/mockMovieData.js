const mockMovieData = {
  results: [
    {
      "vote_count": 2193,
      "id": 346364,
      "video": false,
      "vote_average": 7.4,
      "title": "It",
      "popularity": 810.855615,
      "poster_path": "/9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
      "original_language": "en",
      "original_title": "It",
      "genre_ids": [
        12,
        18,
        27
      ],
      "backdrop_path": "/tcheoA2nPATCm2vvXw2hVQoaEFD.jpg",
      "adult": false,
      "overview": "In a small town in Maine, seven children known as The Losers Club come face to face with life problems.",
      "release_date": "2017-09-05"
    },
    {
      "vote_count": 881,
      "id": 335984,
      "video": false,
      "vote_average": 7.8,
      "title": "Blade Runner 2049",
      "popularity": 452.191852,
      "poster_path": "/aMpyrCizvSdc0UIMblJ1srVgAEF.jpg",
      "original_language": "en",
      "original_title": "Blade Runner 2049",
      "genre_ids": [
        28,
        9648,
        878,
        53
      ],
      "backdrop_path": "/mVr0UiqyltcfqxbAUcLl9zWL8ah.jpg",
      "adult": false,
      "overview": "Thirty years after the events of the first film, a new blade runner, LAPD Officer K, unearths.",
      "release_date": "2017-10-04"
    },
    {
      "vote_count": 1812,
      "id": 281338,
      "video": false,
      "vote_average": 6.7,
      "title": "War for the Planet of the Apes",
      "popularity": 299.274025,
      "poster_path": "/3vYhLLxrTtZLysXtIWktmd57Snv.jpg",
      "original_language": "en",
      "original_title": "War for the Planet of the Apes",
      "genre_ids": [
        18,
        878,
        10752
      ],
      "backdrop_path": "/ulMscezy9YX0bhknvJbZoUgQxO5.jpg",
      "adult": false,
      "overview": "Caesar and his apes are forced into a deadly conflict with an army of humans.",
      "release_date": "2017-07-11"
    },
    {
      "vote_count": 716,
      "id": 343668,
      "video": false,
      "vote_average": 7.3,
      "title": "Kingsman: The Golden Circle",
      "popularity": 257.544191,
      "poster_path": "/pKESfn2Pdy0b7drvZHQb7UzgqoY.jpg",
      "original_language": "en",
      "original_title": "Kingsman: The Golden Circle",
      "genre_ids": [
        28,
        12,
        35
      ],
      "backdrop_path": "/uExPmkOHJySrbJyJDJylHDqaT58.jpg",
      "adult": false,
      "overview": "When an attack on the Kingsman headquarters takes place and a new villain rises, Eggsy and Merlin are forced to work together.",
      "release_date": "2017-09-20"
    },
    {
      "vote_count": 486,
      "id": 381283,
      "video": false,
      "vote_average": 6.9,
      "title": "mother!",
      "popularity": 208.622346,
      "poster_path": "/qmi2dsuoyzZdJ0WFZYQazbX8ILj.jpg",
      "original_language": "en",
      "original_title": "mother!",
      "genre_ids": [
        18,
        27
      ],
      "backdrop_path": "/uuQpQ8VDOtVL2IO4y2pR58odkS5.jpg",
      "adult": false,
      "overview": "A couple's relationship is tested when uninvited guests arrive at their home, disrupting their tranquil existence.",
      "release_date": "2017-09-13"
    },
    {
      "vote_count": 2179,
      "id": 339403,
      "video": false,
      "vote_average": 7.2,
      "title": "Baby Driver",
      "popularity": 179.752717,
      "poster_path": "/dN9LbVNNZFITwfaRjl4tmwGWkRg.jpg",
      "original_language": "en",
      "original_title": "Baby Driver",
      "genre_ids": [
        28,
        80
      ],
      "backdrop_path": "/goCvLSUFz0p7k8R10Hv4CVh3EQv.jpg",
      "adult": false,
      "overview": "After being coerced into working for a crime boss, a young getaway driver finds himself taking part in a heist.",
      "release_date": "2017-06-28"
    },
    {
      "vote_count": 244,
      "id": 415842,
      "video": false,
      "vote_average": 5.1,
      "title": "American Assassin",
      "popularity": 116.888654,
      "poster_path": "/o40BAqdTQHiN3cUfpgieDUYI71z.jpg",
      "original_language": "en",
      "original_title": "American Assassin",
      "genre_ids": [
        28,
        53
      ],
      "backdrop_path": "/puKZWmBIpuEMwGCn2hZkublG1rO.jpg",
      "adult": false,
      "overview": "Following the murder of his fiancée, Mitch Rapp trains under the instruction of Cold War has-been Stan Hurley.",
      "release_date": "2017-09-14"
    },
    {
      "vote_count": 31,
      "id": 440021,
      "video": false,
      "vote_average": 7.3,
      "title": "Happy Death Day",
      "popularity": 116.790332,
      "poster_path": "/FFm2DNpzfPDvIy8DKSyZ71et8W.jpg",
      "original_language": "en",
      "original_title": "Happy Death Day",
      "genre_ids": [
        27,
        9648,
        53
      ],
      "backdrop_path": "/g8RPRNaXqF456tEo2PA6ejrD8Bh.jpg",
      "adult": false,
      "overview": "A college student relives the day of her murder over and over again as she tries to discover her killer's identity.",
      "release_date": "2017-10-12"
    },
    {
      "vote_count": 63,
      "id": 427900,
      "video": false,
      "vote_average": 4.3,
      "title": "Home Again",
      "popularity": 106.54888,
      "poster_path": "/z5CtCxpblBke2G8c7CMkstedMgY.jpg",
      "original_language": "en",
      "original_title": "Home Again",
      "genre_ids": [
        35,
        18
      ],
      "backdrop_path": "/pxpEFY66yYr5bA1N4PCm5b4uz40.jpg",
      "adult": false,
      "overview": "Life for a single mom in Los Angeles takes an expected turn when she allows three young guys to move in with her.",
      "release_date": "2017-09-07"
    },
    {
      "vote_count": 58,
      "id": 395814,
      "video": false,
      "vote_average": 6,
      "title": "Rememory",
      "popularity": 103.16593,
      "poster_path": "/sGQ4kix7bIT35ePpJzA2rNNaxPY.jpg",
      "original_language": "en",
      "original_title": "Rememory",
      "genre_ids": [
        878,
        18,
        9648
      ],
      "backdrop_path": "/3vSrGsnMjzAycIU80udh7NFAJkP.jpg",
      "adult": false,
      "overview": "The widow of a wise professor stumbles upon one of his inventions that's able to record and play a person's memory.",
      "release_date": "2017-09-08"
    },
    {
      "vote_count": 2,
      "id": 446173,
      "video": false,
      "vote_average": 4.5,
      "title": "Ex Libris – New York Public Library",
      "popularity": 93.095023,
      "poster_path": "/5UAtolZFuOFYqUpHcDYamsVKWvb.jpg",
      "original_language": "en",
      "original_title": "Ex Libris – New York Public Library",
      "genre_ids": [
        99
      ],
      "backdrop_path": null,
      "adult": false,
      "overview": "A documentary about how a dominant cultural and demographic institution both sustains their traditional activities and adapts to the digital revolution.",
      "release_date": "2017-09-13"
    },
    {
      "vote_count": 3868,
      "id": 78,
      "video": false,
      "vote_average": 7.9,
      "title": "Blade Runner",
      "popularity": 68.053919,
      "poster_path": "/p64TtbZGCElxQHpAMWmDHkWJlH2.jpg",
      "original_language": "en",
      "original_title": "Blade Runner",
      "genre_ids": [
        878,
        18,
        53
      ],
      "backdrop_path": "/5hJ0XDCxE3qGfp1H3h7HQP9rLfU.jpg",
      "adult": false,
      "overview": "In the smog-choked dystopian Los Angeles of 2019, blade runner Rick Deckard is called out of retirement.",
      "release_date": "1982-06-25"
    },
    {
      "vote_count": 1116,
      "id": 396422,
      "video": false,
      "vote_average": 6.4,
      "title": "Annabelle: Creation",
      "popularity": 66.280927,
      "poster_path": "/tb86j8jVCVsdZnzf8I6cIi65IeM.jpg",
      "original_language": "en",
      "original_title": "Annabelle: Creation",
      "genre_ids": [
        53,
        27
      ],
      "backdrop_path": "/o8u0NyEigCEaZHBdCYTRfXR8U4i.jpg",
      "adult": false,
      "overview": "Several years after the tragic death of their little girl, a dollmaker and his wife welcome a nun.",
      "release_date": "2017-08-03"
    },
    {
      "vote_count": 754,
      "id": 353491,
      "video": false,
      "vote_average": 5.7,
      "title": "The Dark Tower",
      "popularity": 60.255866,
      "poster_path": "/i9GUSgddIqrroubiLsvvMRYyRy0.jpg",
      "original_language": "en",
      "original_title": "The Dark Tower",
      "genre_ids": [
        28,
        37,
        878,
        14,
        27
      ],
      "backdrop_path": "/pVVobDO8cezhVPvwD6EBUN0g3mt.jpg",
      "adult": false,
      "overview": "The last Gunslinger, Roland Deschain, has been locked in an eternal battle.",
      "release_date": "2017-08-03"
    },
    {
      "vote_count": 47,
      "id": 372343,
      "video": false,
      "vote_average": 4.9,
      "title": "The Snowman",
      "popularity": 56.507602,
      "poster_path": "/bQHgpTVsAWjNQWS0frsl7DlzLX1.jpg",
      "original_language": "en",
      "original_title": "The Snowman",
      "genre_ids": [
        80,
        18,
        27,
        9648,
        53
      ],
      "backdrop_path": "/tAg6KUBANIVbYUpTHy5oKibhhw3.jpg",
      "adult": false,
      "overview": "Detective Harry Hole investigates the disappearance of a woman.",
      "release_date": "2017-10-12"
    },
    {
      "vote_count": 168,
      "id": 268531,
      "video": false,
      "vote_average": 6.4,
      "title": "Captain Underpants: The First Epic Movie",
      "popularity": 50.620749,
      "poster_path": "/AjHZIkzhPXrRNE4VSLVWx6dirK9.jpg",
      "original_language": "en",
      "original_title": "Captain Underpants: The First Epic Movie",
      "genre_ids": [
        28,
        16,
        35,
        10751
      ],
      "backdrop_path": "/4F2np8VZjzNJh0SXvVyG4xph9eO.jpg",
      "adult": false,
      "overview": "Two mischievous kids hypnotize their mean elementary school principal.",
      "release_date": "2017-06-01"
    },
    {
      "vote_count": 276,
      "id": 337170,
      "video": false,
      "vote_average": 6.3,
      "title": "American Made",
      "popularity": 48.33967,
      "poster_path": "/8dTWj3c74RDhXfSUZpuyVNJrgS.jpg",
      "original_language": "en",
      "original_title": "American Made",
      "genre_ids": [
        35,
        28,
        80,
        53
      ],
      "backdrop_path": "/jcKzxRGYiXuS7ctTHLdw9wH8d7V.jpg",
      "adult": false,
      "overview": "The true story of pilot Barry Seal, who transported contraband for the CIA and the Medellin cartel in the 1980s.",
      "release_date": "2017-08-17"
    },
    {
      "vote_count": 821,
      "id": 390043,
      "video": false,
      "vote_average": 6.6,
      "title": "The Hitman's Bodyguard",
      "popularity": 44.875789,
      "poster_path": "/5CGjlz2vyBhW5xHW4eNOZIdgzYq.jpg",
      "original_language": "en",
      "original_title": "The Hitman's Bodyguard",
      "genre_ids": [
        28,
        35
      ],
      "backdrop_path": "/7KsqfXDECZMryX1Rv4RKsT7SIjQ.jpg",
      "adult": false,
      "overview": "The world's top bodyguard gets a new client.",
      "release_date": "2017-08-16"
    },
    {
      "vote_count": 344,
      "id": 378236,
      "video": false,
      "vote_average": 5.7,
      "title": "The Emoji Movie",
      "popularity": 44.744654,
      "poster_path": "/f5pF4OYzh4wb1dYL2ARQNdqUsEZ.jpg",
      "original_language": "en",
      "original_title": "The Emoji Movie",
      "genre_ids": [
        35,
        10751,
        16
      ],
      "backdrop_path": "/kTrgxhRSj2sun89bDbnGCPBiey6.jpg",
      "adult": false,
      "overview": "Gene, a multi-expressional emoji, sets out on a journey to become a normal emoji.",
      "release_date": "2017-07-28"
    },
    {
      "vote_count": 166,
      "id": 378064,
      "video": false,
      "vote_average": 8.2,
      "title": "A Silent Voice",
      "popularity": 44.112126,
      "poster_path": "/drlyoSKDOPnxzJFrRWGqzDsyJvR.jpg",
      "original_language": "ja",
      "original_title": "聲の形",
      "genre_ids": [
        16,
        18
      ],
      "backdrop_path": "/geINneHhMbGDeDCGun0elRjVQ1Z.jpg",
      "adult": false,
      "overview": "Shouya Ishida starts bullying the new girl in class.",
      "release_date": "2016-09-17"
    }
  ]
};

export default mockMovieData;
