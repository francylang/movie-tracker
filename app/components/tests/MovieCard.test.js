import React from "react";
import MovieCard from '../MovieCard';
import { shallow } from "enzyme";
import mockMovieData from '../../helpers/mockMovieData';

describe('MovieCard', () => {
  const movie = mockMovieData.results[0];
  const wrapper = shallow(<MovieCard
    movie={movie}
    addFavorites={() => { jest.fn(); }} />);

  it('should render a card', () => {
    const card = wrapper.find('.movie-card');

    expect(card.length).toEqual(1);
  });

  it('should render an img', () => {
    const img = wrapper.find('img');

    expect(img.length).toEqual(2);
    expect(img.first().html().includes(movie.poster_path)).toEqual(true);
  });

  it('should render a release date', () => {
    const release = wrapper.find('.release');

    expect(release.length).toEqual(1);
    expect(release.text()).toEqual(`In theaters: ${movie.release_date}`);
  });

  it('should render a score', () => {
    const score = wrapper.find('.score');

    expect(score.length).toEqual(1);
    expect(score.text()).toEqual(`Score: ${movie.vote_average}`);
  });

  it('should render an overview', () => {
    const overview = wrapper.find('p');

    expect(overview.length).toEqual(1);
    expect(overview.text()).toEqual(movie.overview);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
