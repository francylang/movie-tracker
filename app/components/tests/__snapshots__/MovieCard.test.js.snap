// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    addFavorites={[Function]}
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/tcheoA2nPATCm2vvXw2hVQoaEFD.jpg",
        "genre_ids": Array [
          12,
          18,
          27,
        ],
        "id": 346364,
        "original_language": "en",
        "original_title": "It",
        "overview": "In a small town in Maine, seven children known as The Losers Club come face to face with life problems.",
        "popularity": 810.855615,
        "poster_path": "/9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
        "release_date": "2017-09-05",
        "title": "It",
        "video": false,
        "vote_average": 7.4,
        "vote_count": 2193,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flip-container"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <div
          className="flipper"
        >
          <div
            className="front"
          >
            <div
              className="fav-btn"
              onClick={[Function]}
            />
            <img
              alt="A promotional poster for the movie It"
              className="movie-image"
              src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
            />
          </div>
          <div
            className="back"
          >
            <div
              className="fav-btn"
              onClick={[Function]}
            />
            <img
              alt="A promotional poster for the movie It"
              className="movie-image-back"
              src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
            />
            <div
              className="card-text"
            >
              <h1
                className="movie-title"
              >
                It
              </h1>
              <h2
                className="score"
              >
                Score: 
                7.4
              </h2>
              <p>
                In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
              </p>
              <h4
                className="release"
              >
                In theaters: 
                2017-09-05
              </h4>
            </div>
          </div>
        </div>
      </div>,
      "className": "movie-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flipper"
        >
          <div
            className="front"
          >
            <div
              className="fav-btn"
              onClick={[Function]}
            />
            <img
              alt="A promotional poster for the movie It"
              className="movie-image"
              src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
            />
          </div>
          <div
            className="back"
          >
            <div
              className="fav-btn"
              onClick={[Function]}
            />
            <img
              alt="A promotional poster for the movie It"
              className="movie-image-back"
              src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
            />
            <div
              className="card-text"
            >
              <h1
                className="movie-title"
              >
                It
              </h1>
              <h2
                className="score"
              >
                Score: 
                7.4
              </h2>
              <p>
                In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
              </p>
              <h4
                className="release"
              >
                In theaters: 
                2017-09-05
              </h4>
            </div>
          </div>
        </div>,
        "className": "flip-container",
        "onKeyDown": [Function],
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="front"
            >
              <div
                className="fav-btn"
                onClick={[Function]}
              />
              <img
                alt="A promotional poster for the movie It"
                className="movie-image"
                src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
              />
            </div>,
            <div
              className="back"
            >
              <div
                className="fav-btn"
                onClick={[Function]}
              />
              <img
                alt="A promotional poster for the movie It"
                className="movie-image-back"
                src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
              />
              <div
                className="card-text"
              >
                <h1
                  className="movie-title"
                >
                  It
                </h1>
                <h2
                  className="score"
                >
                  Score: 
                  7.4
                </h2>
                <p>
                  In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                </p>
                <h4
                  className="release"
                >
                  In theaters: 
                  2017-09-05
                </h4>
              </div>
            </div>,
          ],
          "className": "flipper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="fav-btn"
                  onClick={[Function]}
                />,
                <img
                  alt="A promotional poster for the movie It"
                  className="movie-image"
                  src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
                />,
              ],
              "className": "front",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "fav-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "alt": "A promotional poster for the movie It",
                  "className": "movie-image",
                  "src": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="fav-btn"
                  onClick={[Function]}
                />,
                <img
                  alt="A promotional poster for the movie It"
                  className="movie-image-back"
                  src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
                />,
                <div
                  className="card-text"
                >
                  <h1
                    className="movie-title"
                  >
                    It
                  </h1>
                  <h2
                    className="score"
                  >
                    Score: 
                    7.4
                  </h2>
                  <p>
                    In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                  </p>
                  <h4
                    className="release"
                  >
                    In theaters: 
                    2017-09-05
                  </h4>
                </div>,
              ],
              "className": "back",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "fav-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "alt": "A promotional poster for the movie It",
                  "className": "movie-image-back",
                  "src": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="movie-title"
                    >
                      It
                    </h1>,
                    <h2
                      className="score"
                    >
                      Score: 
                      7.4
                    </h2>,
                    <p>
                      In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                    </p>,
                    <h4
                      className="release"
                    >
                      In theaters: 
                      2017-09-05
                    </h4>,
                  ],
                  "className": "card-text",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "It",
                      "className": "movie-title",
                    },
                    "ref": null,
                    "rendered": "It",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Score: ",
                        7.4,
                      ],
                      "className": "score",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Score: ",
                      7.4,
                    ],
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "In a small town in Maine, seven children known as The Losers Club come face to face with life problems.",
                    },
                    "ref": null,
                    "rendered": "In a small town in Maine, seven children known as The Losers Club come face to face with life problems.",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "In theaters: ",
                        "2017-09-05",
                      ],
                      "className": "release",
                    },
                    "ref": null,
                    "rendered": Array [
                      "In theaters: ",
                      "2017-09-05",
                    ],
                    "type": "h4",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flip-container"
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <div
            className="flipper"
          >
            <div
              className="front"
            >
              <div
                className="fav-btn"
                onClick={[Function]}
              />
              <img
                alt="A promotional poster for the movie It"
                className="movie-image"
                src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
              />
            </div>
            <div
              className="back"
            >
              <div
                className="fav-btn"
                onClick={[Function]}
              />
              <img
                alt="A promotional poster for the movie It"
                className="movie-image-back"
                src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
              />
              <div
                className="card-text"
              >
                <h1
                  className="movie-title"
                >
                  It
                </h1>
                <h2
                  className="score"
                >
                  Score: 
                  7.4
                </h2>
                <p>
                  In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                </p>
                <h4
                  className="release"
                >
                  In theaters: 
                  2017-09-05
                </h4>
              </div>
            </div>
          </div>
        </div>,
        "className": "movie-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="flipper"
          >
            <div
              className="front"
            >
              <div
                className="fav-btn"
                onClick={[Function]}
              />
              <img
                alt="A promotional poster for the movie It"
                className="movie-image"
                src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
              />
            </div>
            <div
              className="back"
            >
              <div
                className="fav-btn"
                onClick={[Function]}
              />
              <img
                alt="A promotional poster for the movie It"
                className="movie-image-back"
                src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
              />
              <div
                className="card-text"
              >
                <h1
                  className="movie-title"
                >
                  It
                </h1>
                <h2
                  className="score"
                >
                  Score: 
                  7.4
                </h2>
                <p>
                  In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                </p>
                <h4
                  className="release"
                >
                  In theaters: 
                  2017-09-05
                </h4>
              </div>
            </div>
          </div>,
          "className": "flip-container",
          "onKeyDown": [Function],
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="front"
              >
                <div
                  className="fav-btn"
                  onClick={[Function]}
                />
                <img
                  alt="A promotional poster for the movie It"
                  className="movie-image"
                  src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
                />
              </div>,
              <div
                className="back"
              >
                <div
                  className="fav-btn"
                  onClick={[Function]}
                />
                <img
                  alt="A promotional poster for the movie It"
                  className="movie-image-back"
                  src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
                />
                <div
                  className="card-text"
                >
                  <h1
                    className="movie-title"
                  >
                    It
                  </h1>
                  <h2
                    className="score"
                  >
                    Score: 
                    7.4
                  </h2>
                  <p>
                    In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                  </p>
                  <h4
                    className="release"
                  >
                    In theaters: 
                    2017-09-05
                  </h4>
                </div>
              </div>,
            ],
            "className": "flipper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="fav-btn"
                    onClick={[Function]}
                  />,
                  <img
                    alt="A promotional poster for the movie It"
                    className="movie-image"
                    src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
                  />,
                ],
                "className": "front",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fav-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "A promotional poster for the movie It",
                    "className": "movie-image",
                    "src": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="fav-btn"
                    onClick={[Function]}
                  />,
                  <img
                    alt="A promotional poster for the movie It"
                    className="movie-image-back"
                    src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
                  />,
                  <div
                    className="card-text"
                  >
                    <h1
                      className="movie-title"
                    >
                      It
                    </h1>
                    <h2
                      className="score"
                    >
                      Score: 
                      7.4
                    </h2>
                    <p>
                      In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                    </p>
                    <h4
                      className="release"
                    >
                      In theaters: 
                      2017-09-05
                    </h4>
                  </div>,
                ],
                "className": "back",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fav-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "A promotional poster for the movie It",
                    "className": "movie-image-back",
                    "src": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h1
                        className="movie-title"
                      >
                        It
                      </h1>,
                      <h2
                        className="score"
                      >
                        Score: 
                        7.4
                      </h2>,
                      <p>
                        In a small town in Maine, seven children known as The Losers Club come face to face with life problems.
                      </p>,
                      <h4
                        className="release"
                      >
                        In theaters: 
                        2017-09-05
                      </h4>,
                    ],
                    "className": "card-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "It",
                        "className": "movie-title",
                      },
                      "ref": null,
                      "rendered": "It",
                      "type": "h1",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Score: ",
                          7.4,
                        ],
                        "className": "score",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Score: ",
                        7.4,
                      ],
                      "type": "h2",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "In a small town in Maine, seven children known as The Losers Club come face to face with life problems.",
                      },
                      "ref": null,
                      "rendered": "In a small town in Maine, seven children known as The Losers Club come face to face with life problems.",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "In theaters: ",
                          "2017-09-05",
                        ],
                        "className": "release",
                      },
                      "ref": null,
                      "rendered": Array [
                        "In theaters: ",
                        "2017-09-05",
                      ],
                      "type": "h4",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
